fastchat/train/train.py

```diff
@@ -114,19 +141,21 @@ def preprocess(
         total_len = int(target.ne(tokenizer.pad_token_id).sum())
 
         turns = conversation.split(conv.sep2)
-        cur_len = 1
+        # TODO: make sure the original logic why start with 1?
+        cur_len = 0
         target[:cur_len] = IGNORE_TOKEN_ID
         for i, turn in enumerate(turns):
             if turn == "":
                 break
-            turn_len = len(tokenizer(turn).input_ids)
+            # Fixing the mismatch warning
+            turn_len = len(tokenizer(turn + conv.sep2).input_ids)
 
             parts = turn.split(sep)
             if len(parts) != 2:
                 break
             parts[0] += sep
-            # "-2" is hardcoded for the LLaMA tokenizer to make the offset correct.
-            instruction_len = len(tokenizer(parts[0]).input_ids) - 2
+            # TODO: not clear why -2 is needed, removed here
```

Comment:
This `-2` is required. Please uncomment L137/L166 and print the unmasked tokens. The printed token should only include the outputs generated by the assistant and all of its outputs.
Use this rule to to guide your development for this feature and other models

Reply:
Thank you for the answer. I have done this before. Since it is necessary, then this logic will be rolled back first